### PR TITLE
[Instagram] Extract comments

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This adds support for comments extraction for Instagram

### CODE REVIEW

File: Video by carolinemonteithwilson-BKhrorNBKyV.info.json [hunk 1]
```diff
@@ -0,0 +1 @@
...udcf9\ud83d\udca5\ud83d\udc99\ud83d\udc99\ud83d\udc99@dimitribasil @laura_go_run @soopercooper_", "format_id": "0", "extractor": "Instagram", "comment_count": 26, "display_id": "BKhrorNBKyV", "format": "0 - unknown", "_filename": "Video by carolinemonteithwilson-BKhrorNBKyV.mp4", "webpage_url": "https://www.instagram.com/p/BKhrorNBKyV/?taken-by=carolinemonteithwilson", "id": "BKhrorNBKyV", "title": "Video by carolinemonteithwilson", "thumbnails": [{"url": "https://scontent-amt2-1.cdninstagram.com/t51.2885-15/s1080x1080/e15/fr/14280457_305399146494488_2085348421_n.jpg?ig_cache_key=MTM0MjU0NjA3NTEwMTQ3Mzk0MQ%3D%3D.2", "id": "0"}], "uploader_id": "carolinemonteithwilson", "protocol": "https", "timestamp": 1474263985, "extractor_key": "Instagram", "ext": "mp4", "upload_date": "20160919", "thumbnail": "https://scontent-amt2-1.cdninstagram.com/t51.2885-15/s1080x1080/e15/fr/14280457_305399146494488_2085348421_n.jpg?ig_cache_key=MTM0MjU0NjA3NTEwMTQ3Mzk0MQ%3D%3D.2", "url": "https://scontent-amt2-1.cdninstagram.com/t50.2886-16/14406433_1764201343843456_1242211862_n.mp4", "uploader": "carolinemonteithwilson", "playlist": null, "fulltitle": "Video by carolinemonteithwilson", "like_count": 216}
```
Comment (yan12125) [inline Video by carolinemonteithwilson-BKhrorNBKyV.info.json]: Accidentally commited?

Comment (TRox1972) [inline Video by carolinemonteithwilson-BKhrorNBKyV.info.json]: yes. fixed

File: youtube_dl/extractor/instagram.py [hunk 2]
```diff
@@ -101,6 +101,13 @@ def _real_extract(self, url):
                 uploader_id = media.get('owner', {}).get('username')
                 like_count = int_or_none(media.get('likes', {}).get('count'))
                 comment_count = int_or_none(media.get('comments', {}).get('count'))
+                comments = [{
+                    'author': comment.get('user', {}).get('username'),
+                    'author_id': comment.get('user', {}).get('id'),
+                    'id': comment.get('id'),
+                    'text': comment.get('text'),
+                    'timestamp': int_or_none(comment.get('created_at')),
+                } for comment in media.get('comments', {}).get('nodes', {})]
```
Comment (yan12125) [inline youtube_dl/extractor/instagram.py]: Please add a ```_TEST```.

File: Video by carolinemonteithwilson-BKhrorNBKyV.info.json (hunk 1 above)
Comment (yan12125) [inline Video by carolinemonteithwilson-BKhrorNBKyV.info.json]: Thanks!

File: youtube_dl/extractor/instagram.py (hunk 2 above)
Comment (TRox1972) [inline youtube_dl/extractor/instagram.py]: I've updated previous test cases. Is that ok?

File: youtube_dl/extractor/instagram.py [hunk 3]
```diff
@@ -101,6 +103,13 @@ def _real_extract(self, url):
                 uploader_id = media.get('owner', {}).get('username')
                 like_count = int_or_none(media.get('likes', {}).get('count'))
                 comment_count = int_or_none(media.get('comments', {}).get('count'))
+                comments = [{
+                    'author': comment.get('user', {}).get('username'),
+                    'author_id': comment.get('user', {}).get('id'),
+                    'id': comment.get('id'),
+                    'text': comment.get('text'),
```
Comment (yan12125) [inline youtube_dl/extractor/instagram.py]: Either ```text``` or ```html``` is mandatory.

File: youtube_dl/extractor/instagram.py (hunk 2 above)
Comment (yan12125) [inline youtube_dl/extractor/instagram.py]: Looks good. Thanks.

File: youtube_dl/extractor/instagram.py (hunk 3 above)
Comment (TRox1972) [inline youtube_dl/extractor/instagram.py]: fixed

**Comment by yan12125** [comment]
Merged thanks!

**Pull request merged**